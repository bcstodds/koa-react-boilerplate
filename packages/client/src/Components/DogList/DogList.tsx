function DogList(props: {dogs: any[]}) {
  return (
    <div>
      <h1>Dogs</h1>
      {renderDogs(props.dogs)}
    </div>
  );
}

function renderDogs(dogs: any) {
  return (
    <>
      {dogs.map((dog: any, index: number) => {
        return (
          <div key={index}>
            <h2>
              {dog.name} -- {dog.breed}
            </h2>
            <p>{dog.score}</p>
          </div>
        );
      })}
    </>
  );
}

export default DogList;
