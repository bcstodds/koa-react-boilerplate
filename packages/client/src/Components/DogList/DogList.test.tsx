import axios from 'axios';
import React from 'react';
import TestRenderer from 'react-test-renderer';
import DogList from './DogList';

jest.mock('axios');
const {act} = TestRenderer;

test('DogList can render some dogs', () => {
    const dogs = [
        { name: 'Butler', breed: 'Corgi', score: '10/10' }
    ];

    const component = TestRenderer.create(<DogList dogs={dogs}/>);

    let tree = component!.toJSON();
    expect(tree).toMatchSnapshot();
});