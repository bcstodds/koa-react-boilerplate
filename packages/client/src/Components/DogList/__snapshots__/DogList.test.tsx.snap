// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogList can render some dogs 1`] = `
<div>
  <h1>
    Dogs
  </h1>
  <div>
    <h2>
      Butler
       -- 
      Corgi
    </h2>
    <p>
      10/10
    </p>
  </div>
</div>
`;
